WIP: Warnings

joske opened the pull request:
<!-- Thank you for filing a PR! Help us understand by explaining your changes. Happy contributing! -->

## Motivation

stream line the logging: 
* remove `warn!` where safe
  * safely skipping rounds: this seems totally safe, it even says so in the message
* disambiguate identical logging lines

joske committed:
chore: don't log warning for a slow batch signature

joske committed:
chore: disambiguate logging

joske committed:
chore: safe to ignore when safely skipping round

vbar commented on node/narwhal/src/bft.rs:
```diff
@@ -706,7 +712,10 @@ mod tests {
-        assert_eq!(result.unwrap_err().to_string(), "BFT storage reference is out of sync with the current round");
+        assert_eq!(
+            result.unwrap_err().to_string(),
+            "is_leader_quorum_or_nonleaders_available: BFT storage reference is out of sync with the current round"
```
I wouldn't re-use the same error message - even in test, it's going to confuse people grepping for it...

joske replied:
what do you suggest?

vbar replied:
I'd probably leave it unchanged - then the message is unique, and the test framework will show which test it's failing in...

joske replied:
I don't understand what you mean. This text needs to match the actual error returned.

vbar replied:
Oh, sorry, then it's me who doesn't understand...

vbar commented on node/narwhal/src/bft.rs:
```diff
@@ -792,7 +801,10 @@ mod tests {
-        assert_eq!(result.unwrap_err().to_string(), "BFT storage reference is out of sync with the current round");
+        assert_eq!(
+            result.unwrap_err().to_string(),
+            "update_leader_certificate_to_even_round: BFT storage reference is out of sync with the current round"
```
ditto

vbar replied:
Oh, sorry, then it's me who doesn't understand...

joske committed:
fix: avoid changing program logic when lowering log level

vbar committed:
chore: rephrased warning

old text about "duplicate peer" is confusing, because the condition
can be triggered with duplicate traffic, from a single peer
